product_attribute_manager/wizards/product_variant_attribute_value_wizard.py

```diff
@@ -0,0 +1,255 @@
+            return res
+        active_ids = self.env.context.get("active_ids")
+        variants = self.env[active_model].browse(active_ids)
+        attribute_values = (
+            variants.product_template_attribute_value_ids.product_attribute_value_id
+        )
+        available_attributes = attribute_values.mapped("attribute_id")
+        actions = self._get_actions_from_values(attribute_values)
+        res.update(
+            {
+                "product_ids": [(6, 0, variants.ids)],
+                "attribute_value_ids": [(6, 0, attribute_values.ids)],
+                "available_attribute_ids": [(6, 0, available_attributes.ids)],
+                "attributes_action_ids": [(6, 0, actions.ids)],
+                "product_variant_count": len(variants),
+                "product_template_count": len(variants.mapped("product_tmpl_id")),
+            }
+        )
+        return res
+
+    @api.onchange("filter_attribute_id")
+    def _compute_attributes_action_ids(self):
+        """Update actions according to the attribute to filter on."""
+        for rec in self:
+            actions = self._get_actions_from_values(
+                rec.attribute_value_ids, _filter=rec.filter_attribute_id
+            )
+            rec.attributes_action_ids = [(6, 0, actions.ids)]
+
+    def action_apply(self):
+        for product in self.product_ids:
+            self._action_apply(product)
+
+    def _is_attribute_value_being_used(self, variant_id, attribute_value):
+        """Check if attribute value is still used by a variant."""
+        existing_variants = self.env["product.product"].search(
+            [
+                ("id", "!=", variant_id.id),
+                ("product_tmpl_id", "=", variant_id.product_tmpl_id.id),
+            ],
+        )
+        existing_attributes = existing_variants.mapped(
+            "product_template_attribute_value_ids.product_attribute_value_id"
+        )
+        return attribute_value in existing_attributes
+
+    def _action_apply(self, product):
+        """Update a variant with all the actions set by the user in the wizard."""
+        product_tmpl_av_ids = product.product_template_attribute_value_ids
+        pav_ids = product_tmpl_av_ids.mapped("product_attribute_value_id")
+        pavs_to_clean_by_attr = defaultdict(self.env["product.attribute.value"].browse)
+        for value_action in self.attributes_action_ids:
+            action = value_action.attribute_action
+            if action == "do_nothing":
+                continue
+            pav = value_action.product_attribute_value_id
+            if pav not in pav_ids:
+                continue
+            ptav_ids = product.product_template_attribute_value_ids.filtered(
+                lambda r: r.product_attribute_value_id != pav
+            )
+            if action == "delete":
+                if (
+                    pav.id
+                    in product.product_template_attribute_value_ids.attribute_id.ids
+                ):
```

Comment:
The purpose of declaring variable above (https://github.com/sonhd91/product-variant/pull/1/files#diff-4eda0225fc1c5206fd7fa4094cecf8e8e4cdf57f49b5d8dba3312f525680931bR98) is to reuse it here.
```suggestion
                if (pav.id in product_tmpl_av_ids.attribute_id.ids):
```